code/03_further_clean_demog_data.R

```diff
@@ -0,0 +1,276 @@
+# ---------------------------------------------------------------------------- #
+# Further Clean Demographic Data
+# Author: Jeremy W. Eberle
+# ---------------------------------------------------------------------------- #
+
+# ---------------------------------------------------------------------------- #
+# Notes ----
+# ---------------------------------------------------------------------------- #
+
+# Before running script, restart R (CTRL+SHIFT+F10 on Windows) and set working 
+# directory to parent folder
+
+# ---------------------------------------------------------------------------- #
+# Store working directory, check correct R version, load packages ----
+# ---------------------------------------------------------------------------- #
+
+# Load custom functions
+
+source("./code/01a_define_functions.R")
+
+# Check correct R version, load groundhog package, and specify groundhog_day
+
+groundhog_day <- version_control()
+
+# No packages loaded
+
+# ---------------------------------------------------------------------------- #
+# Import data ----
+# ---------------------------------------------------------------------------- #
+
+load("./data/intermediate_clean_further/dat.RData")
+
+# ---------------------------------------------------------------------------- #
+# Recode "Prefer not to answer" in categorical and ordinal variables ----
+# ---------------------------------------------------------------------------- #
+
+pna <- "Prefer not to answer"
+
+# Recode 555 as "Prefer not to answer". 555 in "birth_year" (integer variable)
+# is recoded after "age" is computed below.
+
+target_vars <- c("education", "employment_stat", "ethnicity", "gender", "income",
+                 "marital_stat")
+dat$demographics[, target_vars][dat$demographics[, target_vars] == 555] <- pna
+
+dat$demographics_race$race[dat$demographics_race$race == 555] <- pna
+
+# Recode "NoAnswer" as "Prefer not to answer"
+
+dat$demographics$country[dat$demographics$country == "NoAnswer"] <- pna
+
+# ---------------------------------------------------------------------------- #
+# Compute age ----
+# ---------------------------------------------------------------------------- #
+
+# Compute age (use NA where "birth_year" is 555 ["prefer not to answer"])
+
+dat$demographics$age <- NA
+
+for (i in 1:nrow(dat$demographics)) {
```

Comment:
Can use `mutate` and `case_when` in situations like this, to avoid loops

Reply:
Thanks! I generally try to use base R but appreciate these tips!